seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/sink/AbstractJdbcSinkWriter.java

```diff
@@ -88,8 +89,23 @@ protected void processSchemaChangeEvent(AlterTableColumnEvent event) throws IOEx
         List<Column> columns = newTableSchema.getColumns();
         switch (event.getEventType()) {
             case SCHEMA_CHANGE_ADD_COLUMN:
-                Column addColumn = ((AlterTableAddColumnEvent) event).getColumn();
-                columns.add(addColumn);
+                AlterTableAddColumnEvent alterTableAddColumnEvent =
+                        (AlterTableAddColumnEvent) event;
+                Column addColumn = alterTableAddColumnEvent.getColumn();
+                String afterColumn = alterTableAddColumnEvent.getAfterColumn();
+                if (StringUtils.isNotBlank(afterColumn)) {
+                    Optional<Column> columnOptional =
+                            columns.stream()
+                                    .filter(column -> afterColumn.equals(column.getName()))
+                                    .findFirst();
+                    columnOptional.ifPresent(
```

Comment:
if not present (basically not happen, but we should care about it), it will lose column?